### PR TITLE
Add Genshiro network

### PR DESCRIPTION
**Pull request by Overseven**

### CODE REVIEW

**Comment by Overseven** [comment]
We need to be able to connect Ledger devices to both networks. 
How to reach this goal if both networks have the same prefix?

**Comment by BenWhiteJam** [comment]
What do you mean by connect Ledger device? You will need to have a Ledger App oder connect through the Polkadot extension to Ledger, no?
How much does the token abbreviation TOKEN make sense for Genshiro, shouldn’t it be more aligned with the project name?

**Comment by Overseven** [comment]
Yep :)
But the polkadot js extension takes the list of networks from here and filters by ledger availability: 
https://github.com/polkadot-js/common/blob/8880e8fa8c715d69a1ba4a7f85f103ffbdbd1bbb/packages/networks/src/interfaces.ts#L55-L64

https://github.com/polkadot-js/common/blob/master/packages/networks/src/substrate.ts#L9-L19

**Comment by Overseven** [comment]
> How much does the token abbreviation TOKEN make sense for Genshiro, shouldn’t it be more aligned with the project name?
Currently our network are using 11 assets and its count will be increased. What is the best solution about symbols naming? Use just one main token name?

**Comment by bkchr** [comment]
>  > How much does the token abbreviation TOKEN make sense for Genshiro, shouldn’t it be more aligned with the project name?
> 
> Currently our network are using 11 assets and its count will be increased. What is the best solution about symbols naming? Use just one main token name?

I would say that the idea was to have all of them in the document, because of this reason the tokens are a list.

**Comment by Overseven** [comment]
> I would say that the idea was to have all of them in the document, because of this reason the tokens are a list.

Well, then it will look like this:

`"symbols": ["GENS", "EQD", "LPT0", "KSM", "BNB", "CRV", "DAI", "ETH", "BUSD", "USDC", "USDT", "WBTC"]`

The first three assets are created by ourselves, others are bridged from other networks. Do I need to set only ours or all assets in the 'symbols' field?

**Comment by Overseven** [comment]
I am more concerned about the requirement for the uniqueness of the prefix for networks. We already have our second project  Equilibrium. By using the Polkadot js extension we can already connect ledger devices to Equilibrium, but with this limitation, we can't add Genshiro, because both networks have the same prefix

**Comment by bkchr** [comment]
> > I would say that the idea was to have all of them in the document, because of this reason the tokens are a list.
> 
> Well, then it will look like this:
> 
> `"symbols": ["GENS", "EQD", "LPT0", "KSM", "BNB", "CRV", "DAI", "ETH", "BUSD", "USDC", "USDT", "WBTC"]`
> 
> The first three assets are created by ourselves, others are bridged from other networks. Do I need to set only ours or all assets in the 'symbols' field?

Wdyt @jacogr ?

**Comment by joepetrowski** [comment]
> The first three assets are created by ourselves, others are bridged from other networks. Do I need to set only ours or all assets in the 'symbols' field?

IMO This should only set the native tokens of the network. Other assets should come with some asset metadata, either on-chain or in your own Equilibrium/Genshiro registry.

**Comment by Overseven** [comment]
Ok, I updated it.
But I still get an error: "failed to generate code from json: prefixes must be unique but this account’s prefix" from "continuous-integration/gitlab-clippy" check

File: ss58-registry.json [hunk 1]
```diff
@@ -496,6 +496,15 @@
       "standardAccount": "*25519",
       "website": "https://equilibrium.io"
     },
+    {
+      "prefix": 67,
+      "network": "genshiro",
+      "displayName": "Genshiro Network",
+      "symbols": ["GENS", "EQD", "LPT0"],
+      "decimals": [9],
```
Comment (joepetrowski) [inline ss58-registry.json:504]: Should be the same length as `symbols`.

**Comment by joepetrowski** [comment]
> Ok, I updated it. But I still get an error: "failed to generate code from json: prefixes must be unique but this account’s prefix" from "continuous-integration/gitlab-clippy" check

Is there a reason that the two networks have the same prefix? The prefixes should make addresses identifiable as belonging to separate networks.

File: ss58-registry.json (hunk 1 above)
Comment (Overseven) [reply ss58-registry.json]: thanks, fixed

**Comment by Overseven** [comment]
> Is there a reason that the two networks have the same prefix? The prefixes should make addresses identifiable as belonging to separate networks.

Because the same address is more simple to use for our users. We have two projects with started mainnets.

**Comment by joepetrowski** [comment]
It's going to be tough to resolve. When I wrote the spec for this registry, I decided that the network name should be [unique to a prefix](https://github.com/paritytech/ss58-registry/blob/0a9e7300cd7c748b21d67fd060054d759a9c7d38/ss58-registry.json#L5). Changing that could have downstream effects on tools that might use network names to identify entries in this registry.

Changing `network` from `string` to `[string]` is definitely something we could evaluate but is outside the scope of this PR. If you want to merge this ASAP, my recommendation is to choose another prefix for Genshiro (68 looks conveniently available). It mostly affects UI rendering and is something that can be explained in wallets.

**Comment by Overseven** [comment]
Ok, here is our solution: we changed the Equilibrium prefix to resolve this problem. Please, review it ASAP

**Comment by joepetrowski** [comment]
LGTM but @BenWhiteJam manages the approvals of these

File: ss58-registry.json [hunk 2]
```diff
@@ -498,6 +498,15 @@
...
+      "symbols": ["GENS", "EQD", "LPT0"],
+      "decimals": [9, 9, 9],
+      "standardAccount": "*25519",
+      "website": "https://genshiro.equilibrium.io"
+    },
+    {
+      "prefix": 68,
       "network": "equilibrium",
       "displayName": "Equilibrium Network",
       "symbols": ["Unknown", "USD", "EQ", "ETH", "BTC", "EOS", "DOT", "CRV"],
```
Comment (BenWhiteJam) [inline ss58-registry.json:512]: @joepetrowski Following your comments, wouldn't it be also recommended to list only chain native tokens for the Equilibrium network then?

**Comment by Overseven** [comment]
@BenWhiteJam done

**Review by bkchr — APPROVED** [review]

**Pull request merged**